src/main/java/io/reactivex/internal/util/BackpressureHelper.java

```diff
@@ -68,4 +70,27 @@ public static long add(AtomicLong requested, long n) {
             }
         }
     }
+    
+    /**
+     * Atomically subtract the given number (positive, not validated) form the target field.
+     * @param requested the target field holding the current requested amount
+     * @param n the produced element count, positive (not validated)
+     * @return the new amount
+     */
+    public static long produced(AtomicLong requested, long n) {
+        for (;;) {
+            long r = requested.get();
+            if (r == Long.MAX_VALUE) {
+                return Long.MAX_VALUE;
+            }
+            long u = r - n;
+            if (u < 0L) {
+                RxJavaPlugins.onError(new IllegalStateException("More produced than requested: " + u));
+                u = 0L;
+            }
+            if (requested.compareAndSet(r, u)) {
+                return r;
```

Comment:
Shouldn't you return `u` here? 

Separate tests for this function would be great, and better var names.